dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/ResourceTableController.java

```diff
@@ -137,11 +148,31 @@ public WebMessage analytics(
     config.setExecutedBy(CurrentUserUtil.getCurrentUserDetails().getUid());
     config.setJobParameters(
         new AnalyticsJobParameters(
-            lastYears, skipTableTypes, skipPrograms, skipResourceTables, skipOutliers));
+            lastYears,
+            skipTableTypes,
+            skipCitusTypes,
+            skipPrograms,
+            skipResourceTables,
+            skipOutliers));
 
     return execute(config);
   }
 
+  private Set<AnalyticsTableType> toAnalyticsTableTypes(Set<String> skipCitusTables) {
+    return CollectionUtils.emptyIfNull(skipCitusTables).stream()
+        .filter(StringUtils::isNotBlank)
+        .map(String::trim)
+        .map(this::toAnalyticsTableType)
+        .collect(Collectors.toSet());
+  }
+
+  private AnalyticsTableType toAnalyticsTableType(String typeName) {
```

Comment:
It's preferrable to alwyas reuse an existing ErrorCode or create a new one if nothing matches.
Example or usage:
```
throw new IllegalQueryException(new ErrorMessage(ErrorCode.E7180));
```